src/engine/QueryExecutionTree.cpp

```diff
@@ -164,7 +165,14 @@ std::shared_ptr<QueryExecutionTree> QueryExecutionTree::createSortedTree(
     return qet;
   }
 
+  // Push down sort into Union.
   QueryExecutionContext* qec = qet->getRootOperation()->getExecutionContext();
+  if (auto unionOperation =
```

Comment:
Does this always perform the sort of the children, or only if they are already presorted?
If so, do we know in the meantime that this is always cheaper?